### PR TITLE
Adding deprecation warning for using package in deployment file

### PR DESCRIPTION
**Pull request by pritidesai**
Closes issue #393 

**Fix 1: Produce WARNING when deployment file has `package` instead of `packages`:**

```
wskdeploy -p /Users/pritidesai/Documents/goworkspace/src/github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/flagstests/
==> Output: 2017/08/30 01:42:19 Unmarshal Openwhisk info from internet
2017/08/30 01:42:19 Using /Users/pritidesai/Documents/goworkspace/src/github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/flagstests/manifest.yaml for deployment 
2017/08/30 01:42:19 The URL is https://openwhisk.ng.bluemix.net/api, selected from .wskprops
2017/08/30 01:42:19 The auth key is set, selected from .wskprops
2017/08/30 01:42:19 The namespace is ****, selected from .wskprops
2017/08/30 01:42:19 WARNING: The package YAML key in deployment file will soon be deprecated. Please use packages instead as described in specifications.
2017/08/30 01:42:19 Deploying package helloWorldFlags ... 
2017/08/30 01:42:20 Done!
2017/08/30 01:42:20 Deploying action helloWorldFlags/greeting ... 
2017/08/30 01:42:20 Done!
Done!
Deploying rule myRule ... Done!
2017/08/30 01:42:20 
Deployment completed successfully.
```

**Fix 2: Update all the deployment files we have, replacing `package` with `packages`:**

Original:

```
application:
  name: sampleApp
  package:
    name: samplePackage
    actions:
```

Updated to:

```
application:
  name: sampleApp
  packages:
    samplePackage:
      actions:
```

**Fix 3: wskdeploy was failing to provide parameters from deployment file while deploying actions, they were inserted as "string" instead of actual value:**

```
/wskdeploy -m tests/src/integration/flagstests/manifest.yaml -d tests/src/integration/flagstests/deployment.yml -i -v
2017/08/30 13:16:27 Unmarshal Openwhisk info from internet
2017/08/30 13:16:27 The URL is https://openwhisk.ng.bluemix.net/api, selected from .wskprops
2017/08/30 13:16:27 The auth key is set, selected from .wskprops
2017/08/30 13:16:27 The namespace is *****, selected from .wskprops
         ____      ___                   _    _ _     _     _
        /\   \    / _ \ _ __   ___ _ __ | |  | | |__ (_)___| | __
   /\  /__\   \  | | | | '_ \ / _ \ '_ \| |  | | '_ \| / __| |/ /
  /  \____ \  /  | |_| | |_) |  __/ | | | |/\| | | | | \__ \   <
  \   \  /  \/    \___/| .__/ \___|_| |_|__/\__|_| |_|_|___/_|\_\ 
   \___\/              |_|

Packages:
Name: helloWorldFlags
    bindings: 

  * action: greeting
    bindings: 
        - place : "string"
        - name : "string"
    annotations: 


Triggers:
* trigger: locationUpdateFlags
    bindings: 
    annotations: 

 Rules
* rule: myRule
    - trigger: locationUpdateFlags
    - action: helloWorldFlags/greeting

Do you really want to deploy this? (y/N): 
```

With this PR:

```
./wskdeploy -m tests/src/integration/flagstests/manifest.yaml -d tests/src/integration/flagstests/deployment.yml -i -v
2017/08/30 13:11:04 Unmarshal Openwhisk info from internet
2017/08/30 13:11:04 The URL is https://openwhisk.ng.bluemix.net/api, selected from .wskprops
2017/08/30 13:11:04 The auth key is set, selected from .wskprops
2017/08/30 13:11:04 The namespace is ****, selected from .wskprops
         ____      ___                   _    _ _     _     _
        /\   \    / _ \ _ __   ___ _ __ | |  | | |__ (_)___| | __
   /\  /__\   \  | | | | '_ \ / _ \ '_ \| |  | | '_ \| / __| |/ /
  /  \____ \  /  | |_| | |_) |  __/ | | | |/\| | | | | \__ \   <
  \   \  /  \/    \___/| .__/ \___|_| |_|__/\__|_| |_|_|___/_|\_\ 
   \___\/              |_|

Packages:
Name: helloWorldFlags
    bindings: 

  * action: greeting
    bindings: 
        - name : "Bernie"
        - place : "Vermont"
    annotations: 


Triggers:
* trigger: locationUpdateFlags
    bindings: 
    annotations: 

 Rules
* rule: myRule
    - trigger: locationUpdateFlags
    - action: helloWorldFlags/greeting

Do you really want to deploy this? (y/N): 
```

### CODE REVIEW

**Comment by pritidesai** [comment]
this is ready to review :)

File: tests/src/integration/validatePackageInDeployment/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+application:
+  name: IntegrationTestPackageInDeployment
+  package:
+    name: TestPackageInDeploymentFile
+    actions:
+      helloworld:
+        inputs:
+          name: Amy Validating Package
```
Comment (mrutkows) [inline tests/src/integration/validatePackageInDeployment/deployment.yaml]: Amy?

Comment (pritidesai) [reply tests/src/integration/validatePackageInDeployment/deployment.yaml]: :)

**Comment by houshengbo** [comment]
@pritidesai @lzbj 
Please do not use log. any more. We need to get rid of it for our message management.

**Pull request merged**